Tests/SourceKitTests/SKLocalSwiftTests.swift

```diff
@@ -217,6 +217,73 @@ final class SKLocalSwiftTests: XCTestCase {
     XCTAssertNotEqual(after, selfDot)
   }
 
+  func testFoldingRanges() {
+    let url = URL(fileURLWithPath: "/a.swift")
+    sk.allowUnexpectedNotification = true
+
+    sk.send(DidOpenTextDocument(textDocument: TextDocumentItem(
+      url: url,
+      language: .swift,
+      version: 12,
+      text:
+    """
+    struct S {
+      //c1
+      //c2
+      /*
+       c3
+      */
+      var abc: Int
+
+      func test(a: Int) {
+        guard a > 0 else { return }
+        self.abc = a
+      }
+    }
+    """)))
```

Comment:
I'd like to see some tests that include a doc comment, and maybe some inline /**/ comments within a line.  If you can find cases where a comment right before or after another range end up with the same start or end point, that would be interesting to have tests for as well.